qiskit/primitives/containers/observables_array.py

```diff
@@ -51,8 +51,6 @@ class ObservablesArray(ShapedMixin):
     """An ND-array of Hermitian observables for an :class:`.Estimator` primitive."""
 
     __slots__ = ("_array", "_shape")
-    ALLOWED_BASIS: str = "IXYZ01+-lr"
-    """The allowed characters in basis strings."""
 
```

Comment:
Note that I removed the option to set a custom basis:
1. It is not used by any qiskit package.
2. It complicates implementation and maintenance.
3. The only usage I can think of would be to signal whether projections are allowed - a Boolean flag is simpler for this purpose.